File: JavaArchitecture/app/src/main/java/com/demo/architecture/injection/module/FragmentModule.java
package com.demo.architecture.injection.module;

import android.app.Activity;
import android.content.Context;
import android.support.v4.app.Fragment;

import com.demo.architecture.injection.ActivityContext;

import dagger.Module;
import dagger.Provides;

/**
 * Created by HeYingXin on 2018/4/17.
 */
@Module
public class FragmentModule{

    private Fragment mFragment;

    public FragmentModule(Fragment fragment){
        mFragment = fragment;
    }

    @Provides
    Fragment provideFragment(){
        return mFragment;
    }
    @Provides
    Activity provideActivity(){
        return mFragment.getActivity();
    }

    @Provides
    @ActivityContext
    Context provideContext(){
        return (Context)mFragment.getActivity();
    }
}

File: kotlin-frist-branch-TODO/app/src/main/java/com/mazouri/mvpkotlin/injection/module/FragmentModule.kt
package com.mazouri.mvpkotlin.injection.module

import android.app.Activity
import android.content.Context
import android.support.v4.app.Fragment
import com.mazouri.mvpkotlin.injection.ActivityContext
import dagger.Module
import dagger.Provides

/**
 * Created by HeYingXin on 2018/4/17.
 */
@Module
class FragmentModule(private val mFragment: Fragment) {

    @Provides
    internal fun provideFragment(): Fragment {
        return mFragment
    }
    @Provides
    internal fun provideActivity(): Activity {
        return mFragment.activity as Activity
    }

    @Provides
    @ActivityContext
    internal fun provideContext(): Context {
        return mFragment.activity as Context
    }
}